source/beamline/CLS/CLSExclusiveStatesControl.cpp

```diff
@@ -0,0 +1,103 @@
+#include "CLSExclusiveStatesControl.h"
+#include "beamline/AMPVControl.h"
+
+CLSExclusiveStatesControl::CLSExclusiveStatesControl(const QString &name, const QString &statusPV, const QString &openPV, const QString &closePV, QObject *parent) :
+	AMExclusiveStatesEnumeratedControl(name, "", parent)
+{
+	setStatusControl(new AMReadOnlyPVControl(QString("%1%2").arg(name).arg("Status"), statusPV, this));
+	addClosedState(new AMSinglePVControl(QString("%1%2").arg(name).arg("Close"), closePV, this), Status::Closed, 1);
+	addOpenState(new AMSinglePVControl(QString("%1%2").arg(name).arg("Open"), openPV, this), Status::Open, 1);
+	addBetweenState(Status::Between);
+}
+
+CLSExclusiveStatesControl::~CLSExclusiveStatesControl()
+{
+
+}
+
+int CLSExclusiveStatesControl::alarmStatus() const
+{
+	return control_->alarmStatus();
+}
+
+int CLSExclusiveStatesControl::alarmSeverity() const
+{
+	return control_->alarmSeverity();
+}
+
+bool CLSExclusiveStatesControl::isOpen() const
+{
+	return (int(value()) == Open);
+}
+
+bool CLSExclusiveStatesControl::isClosed() const
+{
+	return (int(value()) == Closed);
+}
+
+bool CLSExclusiveStatesControl::isBetween() const
+{
+	return (int(value()) == Between);
+}
+
+bool CLSExclusiveStatesControl::setBaseControl(AMControl *newControl)
```

Comment:
There is definitely some overlap here, and you're right that the only real difference is the signal connections.
- The updateStates() needs to be called every time the control value changes, because the control's value also dictates the moving status (Between == moving). This is unique to this particular kind of control, I think.
- Same with the alarmChanged. I think this behavior is unique to this particular kind of control.